File: VB/SearchHighlight/Controllers/HomeController.vb
Imports DevExpress.Web.ASPxTreeList
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace SearchHighlight.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Private db As New Models.NORTHWNDEntities()
		Public Function TreeListPartial(ByVal SearchText As String) As ActionResult
			ViewData("SearchText") = SearchText
			Return PartialView("_TreeListPartial", db.Employees.ToList())
		End Function

		Public Function TreeListCustomPartial(ByVal SearchText As String, ByVal isNewSearch? As Boolean) As ActionResult
			ViewData("IsNewSearch") = isNewSearch
			ViewData("SearchText") = SearchText
			Return PartialView("_TreeListPartial", db.Employees.ToList())
		End Function
	End Class
End Namespace

File: CS/SearchHighlight/Controllers/HomeController.cs
using DevExpress.Web.ASPxTreeList;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace SearchHighlight.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        Models.NORTHWNDEntities db = new Models.NORTHWNDEntities();
        public ActionResult TreeListPartial(string SearchText) {
            ViewData["SearchText"] = SearchText;
            return PartialView("_TreeListPartial", db.Employees.ToList());
        }

        public ActionResult TreeListCustomPartial(string SearchText, bool? isNewSearch) {
            ViewData["IsNewSearch"] = isNewSearch;
            ViewData["SearchText"] = SearchText;
            return PartialView("_TreeListPartial", db.Employees.ToList());
        }
    }
}